diff --git a/test/Makefile.autosetup b/test/Makefile.autosetup
index 2c08780c5..6042d29e0 100644
--- a/test/Makefile.autosetup
+++ b/test/Makefile.autosetup
@@ -742,9 +742,13 @@ CFLAGS	+= -I$(SRCDIR)/test
 
 TEST_BINARY = test/neomutt-test$(EXEEXT)
 
+ifeq (test, $(firstword $(MAKECMDGOALS)))
+	TEST_RUNARGS := $(wordlist 2, $(words $(MAKECMDGOALS)), $(MAKECMDGOALS))
+endif
+
 .PHONY: test
 test: $(TEST_BINARY)
-	$(TEST_BINARY)
+	$(TEST_BINARY) $(TEST_RUNARGS)
 
 $(BUILD_DIRS):
 	$(MKDIR_P) $@
