From 1c76590828ed425152b121ea17994e2abdc07f02 Mon Sep 17 00:00:00 2001
From: Richard Russon <rich@flatcap.org>
Date: Mon, 13 May 2024 08:18:12 +0100
Subject: [PATCH] DEBUG op_print

---
 index/functions.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/index/functions.c b/index/functions.c
index 17c7ef355..c00dd8704 100644
--- a/index/functions.c
+++ b/index/functions.c
@@ -83,6 +83,7 @@
 #include <libintl.h>
 #endif
 #endif
+#include "debug/lib.h"
 
 /// Error message for unavailable functions
 static const char *Not_available_in_this_menu = N_("Not available in this menu");
@@ -2077,6 +2078,11 @@ static int op_prev_entry(struct IndexSharedData *shared, struct IndexPrivateData
  */
 static int op_print(struct IndexSharedData *shared, struct IndexPrivateData *priv, int op)
 {
+#ifdef USE_DEBUG_GRAPHVIZ
+  const struct Expando *c_index_format = cs_subset_expando(shared->sub, "index_format");
+  dump_graphviz_expando_node(c_index_format->node);
+  return FR_SUCCESS;
+#endif
   struct EmailArray ea = ARRAY_HEAD_INITIALIZER;
   ea_add_tagged(&ea, shared->mailbox_view, shared->email, priv->tag_prefix);
   mutt_print_message(shared->mailbox, &ea);
