From 74c7b04e363828e4c4d2e5815f3af05897db62b8 Mon Sep 17 00:00:00 2001
From: Richard Russon <rich@flatcap.org>
Date: Wed, 5 Feb 2025 16:10:51 +0000
Subject: [PATCH] WIP OptNoCurses

---
 color/command.c | 27 ++++++++++-----------------
 1 file changed, 10 insertions(+), 17 deletions(-)

diff --git a/color/command.c b/color/command.c
index 5c6d2ee6d..0be4b1dbf 100644
--- a/color/command.c
+++ b/color/command.c
@@ -45,7 +45,6 @@
 #include "command2.h"
 #include "debug.h"
 #include "dump.h"
-#include "globals.h"
 #include "notify2.h"
 #include "parse_color.h"
 #include "regex4.h"
@@ -207,14 +206,11 @@ static enum CommandResult parse_object(struct Buffer *buf, struct Buffer *s,
 static enum CommandResult parse_uncolor_command(struct Buffer *buf, struct Buffer *s,
                                                 struct Buffer *err, bool uncolor)
 {
-  if (OptNoCurses) // No GUI, so quietly discard the command
+  while (MoreArgs(s))
   {
-    while (MoreArgs(s))
-    {
-      parse_extract_token(buf, s, TOKEN_NO_FLAGS);
-    }
-    return MUTT_CMD_SUCCESS;
+    parse_extract_token(buf, s, TOKEN_NO_FLAGS);
   }
+  return MUTT_CMD_SUCCESS;
 
   parse_extract_token(buf, s, TOKEN_NO_FLAGS);
 
@@ -461,14 +457,11 @@ done:
 enum CommandResult parse_uncolor(struct Buffer *buf, struct Buffer *s,
                                  intptr_t data, struct Buffer *err)
 {
-  if (OptNoCurses) // No GUI, so quietly discard the command
+  while (MoreArgs(s))
   {
-    while (MoreArgs(s))
-    {
-      parse_extract_token(buf, s, TOKEN_NO_FLAGS);
-    }
-    return MUTT_CMD_SUCCESS;
+    parse_extract_token(buf, s, TOKEN_NO_FLAGS);
   }
+  return MUTT_CMD_SUCCESS;
 
   color_debug(LL_DEBUG5, "parse: %s\n", buf_string(buf));
   enum CommandResult rc = parse_uncolor_command(buf, s, err, true);
@@ -491,8 +484,8 @@ enum CommandResult parse_unmono(struct Buffer *buf, struct Buffer *s,
 enum CommandResult parse_color(struct Buffer *buf, struct Buffer *s,
                                intptr_t data, struct Buffer *err)
 {
-  // No GUI, or no colours, so quietly discard the command
-  if (OptNoCurses || (COLORS == 0))
+  // No colours, so quietly discard the command
+  if (COLORS == 0)
   {
     while (MoreArgs(s))
     {
@@ -512,8 +505,8 @@ enum CommandResult parse_color(struct Buffer *buf, struct Buffer *s,
 enum CommandResult parse_mono(struct Buffer *buf, struct Buffer *s,
                               intptr_t data, struct Buffer *err)
 {
-  // No GUI, or colours available, so quietly discard the command
-  if (OptNoCurses || (COLORS != 0))
+  // No colours available, so quietly discard the command
+  if (COLORS != 0)
   {
     while (MoreArgs(s))
     {
