From f03cb2a82584895ed2497e13a2ae34fc06127e7c Mon Sep 17 00:00:00 2001
From: Richard Russon <rich@flatcap.org>
Date: Mon, 20 Sep 2021 15:59:42 +0100
Subject: [PATCH 1/2] undo undo def_color

---
 pager/display.c | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/pager/display.c b/pager/display.c
index 06e611fb2e..eb037f2493 100644
--- a/pager/display.c
+++ b/pager/display.c
@@ -116,7 +116,7 @@ static int comp_syntax_t(const void *m1, const void *m2)
 static void resolve_color(struct MuttWindow *win, struct Line *lines, int line_num,
                           int cnt, PagerFlags flags, int special, struct AnsiColor *ansi)
 {
-  struct AttrColor def_color = { 0 }; /* color without syntax highlight */
+  struct AttrColor *def_color = NULL; /* color without syntax highlight */
   struct AttrColor color = { 0 };     /* final color */
   static struct AttrColor last_color = { 0 }; /* last color set */
   bool search = false;
@@ -145,14 +145,14 @@ static void resolve_color(struct MuttWindow *win, struct Line *lines, int line_n
     m = line_num;
   if (flags & MUTT_PAGER_LOGS)
   {
-    def_color = *(lines[line_num].syntax[0].attr_color);
+    def_color = (lines[line_num].syntax[0].attr_color);
   }
   else if (!(flags & MUTT_SHOWCOLOR))
-    def_color = *simple_colors_get(MT_COLOR_NORMAL);
+    def_color = simple_colors_get(MT_COLOR_NORMAL);
   else if (lines[m].color == MT_COLOR_HEADER)
-    def_color = *lines[m].syntax[0].attr_color;
+    def_color = lines[m].syntax[0].attr_color;
   else
-    def_color = *simple_colors_get(lines[m].color);
+    def_color = simple_colors_get(lines[m].color);
 
   if ((flags & MUTT_SHOWCOLOR) && (lines[m].color == MT_COLOR_QUOTED))
   {
@@ -160,17 +160,17 @@ static void resolve_color(struct MuttWindow *win, struct Line *lines, int line_n
 
     if (qc)
     {
-      def_color = attr_color_copy(qc->attr_color);
+      def_color = qc->attr_color;
 
       while (qc && (qc->prefix_len > cnt))
       {
-        def_color = attr_color_copy(qc->attr_color);
+        def_color = qc->attr_color;
         qc = qc->up;
       }
     }
   }
 
-  color = def_color;
+  color = attr_color_copy(def_color);
   if ((flags & MUTT_SHOWCOLOR) && lines[m].syntax_arr_size)
   {
     matching_chunk = bsearch(&cnt, lines[m].syntax, lines[m].syntax_arr_size,
