From 86e181256bf8cc0ac4b53005da17ba8659d5190b Mon Sep 17 00:00:00 2001
From: Kevin McCarthy <kevin@8t8.us>
Date: Sun, 1 Mar 2020 18:07:02 -0800
Subject: [PATCH] Update send_ctx comment about the cur field.

---
 send.h | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/send.h b/send.h
index ae9922d3..8e9eb076 100644
--- a/send.h
+++ b/send.h
@@ -59,10 +59,7 @@ typedef struct send_ctx
   time_t mtime;
   time_t tempfile_mtime;
 
-  /* Note: cur can't be stored in the send_context when
-   * background editing is added.  This is here for now
-   * just to ease refactoring.
-   */
+  /* Note: cur is set to NULL if the session is backgrounded. */
   HEADER *cur;
   unsigned int has_cur : 1;
   unsigned int is_backgrounded : 1;
-- 
2.26.2

