From a5fa73d3021b831b8fbfee4518af436d955d2c84 Mon Sep 17 00:00:00 2001
From: Richard Russon <rich@flatcap.org>
Date: Tue, 31 Mar 2020 18:23:01 +0100
Subject: [PATCH 24/29] WIP pretty

---
 alias/alias.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/alias/alias.c b/alias/alias.c
index 2520ecb05..e6c4bdc6c 100644
--- a/alias/alias.c
+++ b/alias/alias.c
@@ -483,6 +483,12 @@ retry_name:
   TAILQ_INSERT_TAIL(&Aliases, alias, entries);
 
   mutt_str_copy(buf, C_AliasFile, sizeof(buf));
+  char *pretty = NULL;
+  if (mutt_path2_pretty(buf, HomeDir, &pretty))
+  {
+    mutt_str_strfcpy(buf, pretty, sizeof(buf));
+    FREE(&pretty);
+  }
   if (mutt_get_field(_("Save to file: "), buf, sizeof(buf), MUTT_FILE | MUTT_CLEAR) != 0)
     return;
   mutt_expand_path(buf, sizeof(buf));
